test/tests/functional/pbs_highreslog.py

```diff
@@ -0,0 +1,153 @@
+        if len(lines) > 1:
+            # Remove first line as it is not log line
+            lines = lines[1:]
+        ok = True
+        for line in lines:
+            m = self.tm_micro_re.match(line)
+            if m is not None:
+                ok = False
+        _msg = 'Found high resolution time stamp in tracejob, ' \
+            'it should not be there'
+        self.assertTrue(ok, _msg)
+        _msg = self.server.shortname + \
+            ': High resolution time stamp not found in tracejob output'
+        self.logger.info(_msg)
+
+    def test_basic(self):
+        """
+        Enable High resolution logging, restart server
+        and look for high resolution time stamp in server log
+        """
+        a = {'PBS_LOG_HIGHRES_TIMESTAMP': 1}
+        self.du.set_pbs_config(confs=a, append=True)
+        _msg = 'Failed to restart server: %s' % (self.server.shortname)
+        self.assertTrue(self.server.restart(), _msg)
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+        lines = self.server.log_lines(logtype=self.server, n=20)
+        ok = True
+        for line in lines:
+            m = self.tm_micro_re.match(line)
+            if m is None:
+                ok = False
+        _msg = 'Not Found high resolution time stamp in log'
+        self.assertTrue(ok, _msg)
+        _msg = self.server.shortname + \
+            ': High resolution time stamp found in log'
+        self.logger.info(_msg)
+
+    def test_basic_tracejob(self):
+        """
+        Enable High resolution logging, restart PBS Daemons
+        and look for high resolution time stamp in tracejob output
+        """
+        a = {'PBS_LOG_HIGHRES_TIMESTAMP': 1}
+        self.du.set_pbs_config(confs=a, append=True)
+        PBSInitServices().restart()
+        self.assertTrue(self.server.isUp(), 'Failed to restart PBS Daemons')
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+        lines = self.server.log_lines(logtype='tracejob', id=jid, n='ALL')
+        if len(lines) > 1:
+            # Remove first line as it is not log line
+            lines = lines[1:]
+        ok = True
+        for line in lines:
+            m = self.tm_micro_re.match(line)
+            if m is None:
+                # If this is accounting log, ignore it as
+                # Accounting log does not have high res logging
+                line = line.split()
+                if (len(line) >= 3) and (line[2].strip() != 'A'):
+                    ok = False
+        _msg = 'Not Found high resolution time stamp in log'
+        self.assertTrue(ok, _msg)
+        _msg = self.server.shortname + \
+            ': High resolution time stamp found in log'
+        self.logger.info(_msg)
+
+    def tearDown(self):
+        confs = self.du.parse_pbs_config()
+        if 'PBS_LOG_HIGHRES_TIMESTAMP' in confs:
```

Comment:
Shouldn't this be added as part of default setup? why are you removing it here?

Reply:
The problem is that pbs.conf is not reverted to default in the default setUp().  It needs to be removed here.